addons/binding/org.openhab.binding.smartthings/src/main/java/org/openhab/binding/smartthings/internal/SmartthingsHandlerFactory.java

```diff
@@ -0,0 +1,185 @@
+/**
+ * Copyright (c) 2010-2019 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.smartthings.internal;
+
+import static org.openhab.binding.smartthings.SmartthingsBindingConstants.*;
+
+import java.util.ArrayList;
+import java.util.Dictionary;
+import java.util.HashMap;
+import java.util.Hashtable;
+import java.util.List;
+import java.util.Map;
+
+import org.eclipse.smarthome.config.discovery.DiscoveryService;
+import org.eclipse.smarthome.core.thing.Bridge;
+import org.eclipse.smarthome.core.thing.Channel;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingTypeUID;
+import org.eclipse.smarthome.core.thing.ThingUID;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandlerFactory;
+import org.eclipse.smarthome.core.thing.binding.ThingHandler;
+import org.eclipse.smarthome.core.thing.type.ChannelTypeRegistry;
+import org.openhab.binding.smartthings.discovery.SmartthingsDiscoveryService;
+import org.openhab.binding.smartthings.handler.SmartthingsBridgeHandler;
+import org.openhab.binding.smartthings.handler.SmartthingsThingHandler;
+import org.osgi.framework.ServiceRegistration;
+import org.osgi.service.event.Event;
+import org.osgi.service.event.EventHandler;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.Gson;
+
+/**
+ * The {@link SmartthingsHandlerFactory} is responsible for creating things and thing
+ * handlers.
+ *
+ * @author Bob Raker - Initial contribution
+ */
+public class SmartthingsHandlerFactory extends BaseThingHandlerFactory implements EventHandler {
+
+    private Logger logger = LoggerFactory.getLogger(SmartthingsHandlerFactory.class);
+    private Map<ThingUID, ServiceRegistration<?>> discoveryServiceRegs = new HashMap<>();
+
+    private SmartthingsBridgeHandler bridge;
+    private ChannelTypeRegistry channelTypeRegistry;
+    private Gson gson;
+    private List<SmartthingsThingHandler> thingHandlers = new ArrayList<SmartthingsThingHandler>();
+
+    @Override
+    public boolean supportsThingType(ThingTypeUID thingTypeUID) {
+        return THING_TYPE_SMARTTHINGS.equals(thingTypeUID) || SUPPORTED_THING_TYPES_UIDS.contains(thingTypeUID);
+    }
+
+    public SmartthingsHandlerFactory() {
+        // Get a Gson instance
+        gson = new Gson();
```

Comment:
Ideally, you create only one gson instance for your binding here in the factory and hand it over to the thinghandlers.